server/graphql/common/expenses.ts

```diff
@@ -1475,6 +1513,22 @@ export const DRAFT_EXPENSE_FIELDS = [
   'invoiceInfo',
 ] as const;
 
+/**
+ * Returns true if the value has changed.
+ *
+ * /!\ We have no 1-to-1 field mapping between `Expense` and `ExpenseData`, but since we used to only
+ * check `isNil` this won't introduce any regression. Ideally, this helper should do a mapping between the two.
+ */
+const isValueChanging = (expense: Expense, expenseData: ExpenseData, key: string): boolean => {
+  const nullableFields = ['accountingCategory'];
+  const value = expenseData[key];
+  if (!nullableFields.includes(key)) {
```

Comment:
🍨 This conditional would be simpler to read if the primary conditional wasn't a negation:
```suggestion
  if (nullableFields.includes(key)) {
```

Reply:
True